test/e2e/app-dir/metadata/metadata.test.ts

```diff
@@ -385,31 +385,15 @@ describe('app dir - metadata', () => {
         'og:image:height': '114',
         'og:image:type': 'image/png',
         'og:image:alt': 'A alt txt for og',
-        'og:image': isNextDev
-          ? expect.stringMatching(
-              /http:\/\/localhost:\d+\/opengraph\/static\/opengraph-image/
-            )
-          : expect.stringMatching(
-              new RegExp(
-                `https:\\/\\/(${
-                  isNextDeploy ? '.*?\\.vercel\\.app' : 'example\\.com'
-                })\\/opengraph\\/static\\/opengraph-image`
-              )
-            ),
+        'og:image': expect.stringContaining(
+          'https://example.com/opengraph/static/opengraph-image'
```

Comment:
The test case was also mean to cover the dev/prod/deployment for static images without metadataBase, since in dev the local metadata images should always be served in localhost. If we're requesting them from remote then you'll not viewing the latest local changes in dev. So it's bit unexpected now.

For any random url it makes sense to me if it's always resolved with `metadataBase`;
For the local metadata images, they should be use local in dev, and resolved with metadataBase in build.

Reply:
I'm curious why in dev we'd serve from localhost, if the user is explicitly providing a base URL for the metadata? I think it only makes sense when it's not provided, as described in the docs.

Reply:
I've updated this to not require any changes to tests: the behavior is that Vercel preview deployments will continue to respect the fallback value when using the `opengraph-image` convention so that previews / localhost automagically get the right URL. However, any user provided images via the metadata config will always respect the `metadataBase` value.